model_stubs/BaseAutoencoder.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+
+from torch import nn
+
+
+'''
+Every implemented Autoencoder should inherit this base class
+'''
+
+class BaseAutoencoder(nn.Module):
+
+    def __init__(self):
+        super(BaseAutoencoder, self).__init__()
+
+    @abstractmethod
```

Comment:
Das funktioniert so nicht mit dem `abstractmethod`. Du musst `BaseAutoEncoder` zusätzlich noch von `abc.ABC` erben lassen, erst dann wird überprüft ob die abstrakten Methoden in den Unterklassen implementiert sind oder nicht

```suggestion
from abc import abstractmethod
import abc

from torch import nn


'''
Every implemented Autoencoder should inherit this base class
'''

class BaseAutoencoder(nn.Module, abc.ABC):

    def __init__(self):
        nn.Module.__init__(self)

    @abstractmethod
```